src/network.rs

```diff
@@ -776,6 +776,15 @@ impl<'a> Network<'a> {
     if let Ok((albums, top_tracks, related_artist)) = try_join!(albums, top_tracks, related_artist)
     {
       let mut app = self.app.lock().await;
+
+      app.dispatch(IoEvent::CurrentUserSavedAlbumsContains(
```

Comment:
@Rigellute I'm curious on your opinion here. If you look at the initial commit for this patch, you'll see I took the more direct approach of calling `current_user_saved_albums_contains()` here to be more direct.

I later changed it to dispatch a `IoEvent::CurrentUserSavedAlbumsContains` because this seemed more robust and consistent, and also avoids the brittleness of the prior approach where we had to carefully call `current_user_saved_albums_contains()`` prior to when we take the lock here.

Reply:
The dispatch won't block whereas we'd have to `await` the `current_user_saved_albums_contains` call which could take some, so dispatching the event is the right move to make here to prevent any blocking in this function due to network IO latency.

Reply:
Thanks, I am new to rust so I didn't feel confident.